core/src/main/java/io/grpc/transport/AbstractClientStream.java

```diff
@@ -125,8 +130,8 @@ protected void inboundHeadersReceived(Metadata.Headers headers) {
    *
    * @param frame the received data frame. Its ownership is transferred to this method.
    */
-  protected void inboundDataReceived(ReadableBuffer frame) {
-    Preconditions.checkNotNull(frame, "frame");
```

Comment:
1. My point was just that the exception, with the message, identifies the issue independent of gRPC version. There are many different places that exceptions can propagate out of the library; making sure all of them wrap the exception to include version information is 1) unreliable, 2) a major effort, and 3) dangerous, as it would likely change the type of exception.

2. I'm okay with using a sentence fragment instead. Since these `checkNotNull`s are lots of places, I do find it reasonable to use just the parameter name, since that is generally enough information. If certain places it seems necessary to add more information, I'm okay with that. But I don't want to increase the burden required for the string in all places because that is how you get people arguing to not include the string at all. Simply duplicating the parameter name is effective, easy, and is a pattern that can be reasonably be followed without deep thought.

I'm not worried about line-wrapping checkNotNull. I was using the full form on a project with 80 column wrapping and I don't recall it every being a problem. Do note that I'm not suggesting you to use the full form. I'm simply trying to argue that the original author gets to decide and we don't change it later without cause, as it is a style decision well within their right.

A strong percentage of merges I've had to do were regarding such import changes. Even if you aren't reviewing them, such line churn can create needless work, and I was tired of that needless work for the code I write. Again, it is a personal style decision, and the decision is left to the code author.